subscriber.py

```diff
@@ -23,14 +25,14 @@ def configure(self, context):
         super().configure(context)
         self._subscriber = NioSubscriber(self._subscriber_handler,
                                          topic=self.topic())
-
-    def start(self):
-        """ Start the block by opening the underlying subscriber """
-        super().start()
-        self._subscriber.open()
+        self._subscriber.open(on_connected=self.conn_on_connected,
+                              on_disconnected=self.conn_on_disconnected)
+        # let connectivity configure
+        self.conn_configure(self._subscriber.is_connected)
 
     def stop(self):
         """ Stop the block by closing the underlying subscriber """
+        self.conn_stop()
```

Comment:
My only comment on these, is allowing this to happen in the `super()` allowing the comm-mixin to handleit's configure and close.  However, it almost seems like we want to stop it, before closing the subscriber, then continuing the stop.  So I'm not overly fussy either way on these.

Reply:
My guess would be you'd want to tell the mixin we are stopping before we do the actual stop. Otherwise the closing of the publisher/subscriber would issue the disconnection callback and put the block into warning state.

Reply:
Right, it'd be like wanting to call the mixin super, then do some block work, then continue up the block super.  But I don't think that's very possible.  Unless, the Conn mixin didn't itself call a `super().stop` and in the block we have the two calls `super(PubSubConnectivity, self).stop()`, then `super(GeneratorBlock, self).stop()`, but that's probably getting a bit on the messy side.